shelljs/shelljs.d.ts

```diff
@@ -511,6 +511,18 @@ declare module "shelljs"
      */
     export function error(): string;
 
+
+    export function touch(...files: string[]): void;
+    export function touch(files: string[]): void;
+
+    export function touch(options: string, ...files: string[]): void;
+    export function touch(options: string, files: string[]): void;
+    
+    export function touch(options: Object, ...files: string[]): void;
+    export function touch(options: Object, files: string[]): void;
```

Comment:
We could be more precise concerning the options. They should be restricted to these string literals:
https://github.com/shelljs/shelljs#touchoptions-file_array

E.g.
```TypeScript
type TouchOptions = "-a" |  "-c" | "-m" | "-d" | "-r";
```